--- ./build.gradle.kts	2022-05-02 02:20:38.364208995 -0400
+++ -	2022-06-20 23:05:39.504022789 -0400
@@ -3,6 +3,7 @@
     id("kotlin-android")
     id("kotlin-parcelize")
     kotlin("plugin.serialization")
+    id("com.google.devtools.ksp") version "1.6.20-1.0.5"
 }
 
 android {
@@ -21,13 +22,17 @@
             useSupportLibrary = true
         }
 
-        buildConfigField("int", "DISCORD_VERSION_CODE", "89108")
+        buildConfigField("int", "DISCORD_VERSION_CODE", "124012")
+        buildConfigField("String", "DISCORD_VERSION_NAME", "\"124.12 - Stable\"")
     }
 
     buildTypes {
-        getByName("release") {
+        release {
             isMinifyEnabled = true
-            proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
+            proguardFiles(
+                getDefaultProguardFile("proguard-android-optimize.txt"),
+                "proguard-rules.pro"
+            )
         }
     }
 
@@ -48,7 +53,7 @@
             applicationIdSuffix = ".fosscord"
             versionNameSuffix = "-fosscord"
 
-            buildConfigField("String", "URL_API", "\"https://api.fosscord.com/v9\"")
+            buildConfigField("String", "URL_API", "\"https://api.fosscord.com/api/v9\"")
             buildConfigField("String", "URL_CDN", "\"https://cdn.fosscord.com\"")
             buildConfigField("String", "URL_GATEWAY", "\"wss://gateway.fosscord.com/?v=9&encoding=json&compress=zlib-stream\"")
         }
@@ -66,7 +71,7 @@
                 "-Xopt-in=androidx.compose.animation.ExperimentalAnimationApi" +
                 "-Xopt-in=androidx.compose.material.ExperimentalMaterialApi" +
                 "-Xopt-in=androidx.compose.material3.ExperimentalMaterial3Api" +
-                "-Xcontent-receivers"
+                "-Xcontext-receivers"
     }
 
     buildFeatures {
@@ -82,6 +87,18 @@
             excludes += "/META-INF/{AL2.0,LGPL2.1}"
         }
     }
+
+    sourceSets {
+        applicationVariants.all {
+            getByName(name) {
+                kotlin.srcDir("build/generated/ksp/$name/kotlin")
+            }
+        }
+    }
+
+    kotlin {
+
+    }
 }
 
 dependencies {
@@ -89,15 +106,22 @@
     implementation(project(":overlapping-panels-compose"))
     implementation(project(":simpleast-compose"))
 
+    implementation(project(":partialgen"))
+    ksp(project(":partialgen"))
+
+    implementation(project(":enumgetter"))
+    ksp(project(":enumgetter"))
+
     Dependencies.Koin(this)
     Dependencies.Ktor(this)
-    Dependencies.KotlinXDatetime(this)
+    Dependencies.KotlinX(this)
     Dependencies.HCaptcha(this)
     Dependencies.AndroidxCore(this)
     Dependencies.AndroidxPreferences(this)
     Dependencies.Material(this)
     Dependencies.Compose(this)
     Dependencies.Accompanist(this)
+    Dependencies.Shimmer(this)
     Dependencies.Coil(this)
     Dependencies.ExoPlayer(this)
 }
